subsys/bluetooth/host/hci_core.c

```diff
@@ -1381,27 +1381,30 @@ static void le_remote_feat_complete(struct net_buf *buf)
 static void le_data_len_change(struct net_buf *buf)
 {
 	struct bt_hci_evt_le_data_len_change *evt = (void *)buf->data;
+	uint16_t handle = sys_le16_to_cpu(evt->handle);
+	struct bt_conn *conn;
+
+#if defined(CONFIG_BT_USER_DATA_LEN_UPDATE)
 	uint16_t max_tx_octets = sys_le16_to_cpu(evt->max_tx_octets);
 	uint16_t max_rx_octets = sys_le16_to_cpu(evt->max_rx_octets);
 	uint16_t max_tx_time = sys_le16_to_cpu(evt->max_tx_time);
 	uint16_t max_rx_time = sys_le16_to_cpu(evt->max_rx_time);
-	uint16_t handle = sys_le16_to_cpu(evt->handle);
-	struct bt_conn *conn;
+#endif
```

Comment:
This could even be moved into top of scope of the `#if defined(CONFIG_BT_USER_DATA_LEN_UPDATE)` block starting on line 1400.